src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutItem.cs

```diff
@@ -73,5 +139,708 @@ internal void InvokeClick()
+				object spCommandParameter;
+				bool canExecute;
+
+				spCommandParameter = CommandParameter;
+				canExecute = spCommand.CanExecute(spCommandParameter);
+				if (canExecute)
+				{
+					spCommand.Execute(spCommandParameter);
+				}
+			}
+		}
+
+		// PointerEnter event handler.
+		protected override void OnPointerEntered(PointerRoutedEventArgs pArgs)
+		{
+			base.OnPointerEntered(pArgs);
+
+			MenuFlyoutPresenter spParentPresenter;
+
+			m_bIsPointerOver = true;
+
+			spParentPresenter = GetParentMenuFlyoutPresenter();
+			if (spParentPresenter != null)
+			{
+				IMenuPresenter subPresenter;
+
+				subPresenter = (spParentPresenter as IMenuPresenter).SubPresenter;
+				if (subPresenter != null)
+				{
+					ISubMenuOwner subPresenterOwner;
+
+					subPresenterOwner = subPresenter.Owner;
+
+					if (subPresenterOwner != null)
+					{
+						subPresenterOwner.DelayCloseSubMenu();
+					}
+				}
+			}
+
+			UpdateVisualState();
+		}
+
+		// PointerExited event handler.
+		protected override void OnPointerExited(PointerRoutedEventArgs pArgs)
+		{
+			base.OnPointerExited(pArgs);
+
+			// MenuFlyoutItem does not capture pointer, so PointerExit means the item is no longer pressed.
+			m_bIsPressed = false;
+			m_bIsPointerLeftButtonDown = false;
+			m_bIsPointerOver = false;
+			UpdateVisualState();
+		}
+
+		// PointerCaptureLost event handler.
+		protected override void OnPointerCaptureLost(PointerRoutedEventArgs pArgs)
+		{
+			base.OnPointerCaptureLost(pArgs);
+
+			// MenuFlyoutItem does not capture pointer, so PointerCaptureLost means the item is no longer pressed.
+			m_bIsPressed = false;
+			m_bIsPointerLeftButtonDown = false;
+			m_bIsPointerOver = false;
+			UpdateVisualState();
+		}
+
+		// Called when the IsEnabled property changes.
+		protected override void OnIsEnabledChanged(bool oldValue, bool newValue)
+		{
+			bool bIsEnabled = IsEnabled;
+
+			if (!bIsEnabled)
+			{
+				ClearStateFlags();
+			}
+			else
+			{
+				UpdateVisualState();
+			}
+		}
+
+		// Called when the control got focus.
+		protected override void OnGotFocus(RoutedEventArgs pArgs)
+		{
+			UpdateVisualState();
+		}
+
+		// LostFocus event handler.
+		protected override void OnLostFocus(RoutedEventArgs pArgs)
+		{
+			if (!m_bIsPointerLeftButtonDown)
+			{
+				m_bIsSpaceOrEnterKeyDown = false;
+				m_bIsNavigationAcceptOrGamepadAKeyDown = false;
+				m_bIsPressed = false;
+			}
+
+			UpdateVisualState();
+		}
+
+		// KeyDown event handler.
+		protected override void OnKeyDown(KeyRoutedEventArgs pArgs)
+		{
+			bool handled = false;
+
+			handled = pArgs.Handled;
+			if (!handled)
+			{
+				VirtualKey key = VirtualKey.None;
+
+				key = pArgs.Key;
+				handled = KeyPressMenuFlyout.KeyDown(key, this);
+				pArgs.Handled = handled;
+			}
+		}
+
+		// KeyUp event handler.
+		protected override void OnKeyUp(KeyRoutedEventArgs pArgs)
+		{
+			bool handled = false;
+
+			handled = (pArgs.Handled);
+			if (!handled)
+			{
+				VirtualKey key = VirtualKey.None;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'key'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)